File: src/test/java/upbrella/be/util/AESTest.java
package upbrella.be.util;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

@SpringBootTest
public class AESTest {

    @Autowired
    private AesEncryptor aesEncryptor;

    @Test
    public void testDecrypt() {
        String plainText = "Hello World!";
        String encryptedText = aesEncryptor.encrypt(plainText);
        String decryptedText = aesEncryptor.decrypt(encryptedText);
        assertEquals(plainText, decryptedText); // 원래의 평문과 복호화된 텍스트가 동일함을 확인
    }

    @Test
    public void testEncryptNullInput() {
        String encryptedText = aesEncryptor.encrypt(null);
        assertNull(encryptedText); // 입력이 null인 경우 출력도 null이어야 함을 확인
    }

    @Test
    public void testDecryptNullInput() {
        String decryptedText = aesEncryptor.decrypt(null);
        assertNull(decryptedText); // 입력이 null인 경우 출력도 null이어야 함을 확인
    }
}

File: src/test/kotlin/upbrella/be/util/AesTest.kt
package upbrella.be.util

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class AesTest @Autowired constructor(
    private val aesEncryptor: AesEncryptor,
){
    @Test
    @DisplayName("복호화 테스트")
    fun testDecrypt() {
        // given
        val plainText = "Hello World!"
        val encryptedText = aesEncryptor.encrypt(plainText)

        // when
        val decryptedText = aesEncryptor.decrypt(encryptedText)

        // then
        assertThat(decryptedText).isEqualTo("Hello World!")
    }

    @Test
    @DisplayName("Null을 암호화하는 경우 출력도 Null")
    fun testEncryptNullInput() {
        // given
        val nullString : String? = null

        // when
        val encryptedText = aesEncryptor.encrypt(nullString)

        // then
        assertThat(encryptedText).isNull()
    }

    @Test
    @DisplayName("Null을 복호화 하는 경우 경우 출력도 Null")
    fun testDecryptNullInput() {
        // given
        val nullString : String? = null

        // when
        val decryptedText = aesEncryptor.decrypt(nullString)

        // then
        assertThat(decryptedText).isNull()
    }
}
